lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -121,12 +121,13 @@ IRGenDebugInfo::IRGenDebugInfo(const IRGenOptions &Opts,
   }
 
   unsigned Lang = llvm::dwarf::DW_LANG_Swift;
-  std::string Producer = version::getSwiftFullVersion();
+  std::string Producer = version::getSwiftFullVersion(
+    Opts.EffectiveLanguageVersion);
   bool IsOptimized = Opts.Optimize;
   StringRef Flags = Opts.DWARFDebugFlags;
-  unsigned Major, Minor;
-  std::tie(Major, Minor) = version::getSwiftNumericVersion();
-  unsigned MajorRuntimeVersion = Major;
+
+  assert(!Opts.EffectiveLanguageVersion.empty());
```

Comment:
I don't know what the version is supposed to mean here... is it the compiler version (because clients want to make sure they're working with a compatible compiler) or is it the language dialect? I suspect the former is actually more useful for a debugger.

... but if we do need to use the `EffectiveLanguageVersion`, you can get it off the corresponding Swift module or `ASTContext`.